oaknational/Oak-Web-Application
# feat: connect share to lesson overview - LESQ-193 and LESQ-469

## Description

Music year: 1995

- Adds share buttons ('all' is at top of lesson overview) and on containers
- Refactors some shared types and improved naming for legacy lessons.
- adds tracking and tests for share all button

figma - https://www.figma.com/file/3i0y2PHT6pECaGgvLhoN8z/Lesson-page-July-2023?type=design&node-id=829-60892&mode=dev
ticket - https://www.notion.so/oaknationalacademy/I-want-a-link-to-share-page-from-lesson-overview-page-966a62a629c846908a0503ec8e89ca19



## How to test

1. Go to https://deploy-preview-2096--oak-web-application.netlify.thenational.academy

Go to legacy lesson page - you should be able to share
Go to lesson page - share should be disabled (all button) and not visible on containers

Check that avo and post hog are receiving correct tracking events 
lessonShareStarted
downloadResourceButtonClicked

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist
